jdaviz/core/helpers.py

```diff
@@ -273,96 +274,94 @@ def get_model_parameters(self, models=None, model_label=None, x=None, y=None):
+        scar : sidecar or `None`
+            The ``sidecar.Sidecar`` object used to create the tab or window or
+            None if no sidecar was created.
 
         Notes
         -----
-        If this method is called in the "classic" Jupyter notebook, the app will
-        appear inline, as only lab has a mechanism to have multiple tabs.
-        See Also
-        --------
-        show_in_new_tab
-        show_inline
+        If "sidecar" is requested in the "classic" Jupyter notebook, the app will appear inline,
+        as only Lab has a mechanism to have multiple tabs.
         """
-        if 'title' not in kwargs:
-            kwargs['title'] = self.app.config
+        try:
+            if loc == "inline":
+                display(self.app)
 
-        scar = Sidecar(**kwargs)
-        with scar:
-            display(self.app)
+            elif loc.startswith('sidecar'):
+                # Use default behavior if loc is exactly 'sidecar', else split anchor from the arg
+                anchor = None if loc == 'sidecar' else loc.split(':')[1]
 
-        return scar
-    show_in_sidecar.__doc__ = show_in_sidecar.__doc__.format(
-        sidecar_anchor_values=repr(Sidecar.anchor.values)[1:-1])
+                # If title unset, default to the viz config
+                title = self.app.config if title is None else title
 
-    def show_in_new_tab(self, **kwargs):
-        """
-        Display the Jdaviz application in a new tab in JupyterLab.
+                scar = Sidecar(anchor=anchor, title=title)
+                with scar:
+                    display(self.app)
 
-        Additional keywords not listed here are passed into the
-        ``sidecar.Sidecar`` constructor. See
-        `jupyterlab-sidecar <https://github.com/jupyter-widgets/jupyterlab-sidecar>`_
-        for the most up-to-date options.
+                return scar
 
-        Parameters
-        ----------
-        title : str, optional
-            The title of the sidecar tab.  Defaults to the name of the
-            application; e.g., "specviz".
+            elif loc == "new browser tab":
+                raise NotImplementedError
 
-        Returns
-        -------
-        sidecar
-            The ``sidecar.Sidecar`` object used to create the tab.
+            elif loc == "popout":
+                raise NotImplementedError
 
-        Notes
-        -----
-        If this method is called in the "classic" Jupyter notebook, the app will
-        appear inline, as only lab has a mechanism to have multiple tabs.
+            else:
+                raise ValueError("Unrecognized display location: " + str(loc))
 
-        See Also
-        --------
-        show_in_sidecar
-        show_inline
+        except Exception as e:
+            warnings.warn(f'Error detected in display: {repr(e)}. Falling back to IPython display',
+                          RuntimeWarning)
+            display(self.app)
+
+    def show_in_sidecar(self, anchor=None, title=None):
```

Comment:
Do we want to deprecate this?

Reply:
See https://github.com/spacetelescope/jdaviz/pull/965#discussion_r913820210